core/src/main/javascript/graph/dagger/render/nodesAndEdges.js

```diff
@@ -277,6 +277,13 @@ const tagBulletVerticalOffset = ({ height }) => {
     startOffset + tagIndex * (bulletSize + spaceBetweenBullets);
 };
 
+// join words by whitespace, unless it's delimited by -_
+const joinWords = (words: Array<string>): string =>
+  reduce(
+    words,
+    (acc, word) => (/[-_]$/.test(acc) ? acc + word : acc + " " + word)
```

Comment:
Shouldn't the regex test `word` instead of the accumulator?

Reply:
["this-","is-","it"] consider this sentence, you want to check the suffix of the accumulated result 
 then append each word (the first word being the initial acc)